include/ideep/operators/conv.hpp

```diff
@@ -1928,18 +1928,21 @@ struct convolution_forward
     auto& expected_src = reorder_src ?
         src.reorder_if_differ_in(pd.src_desc()) :
         src;
+    if (reorder_src) {
+      dst.reinit_if_possible(pd.dst_desc());
+    }
     // make sure other has same format with dst.
-    // TODO: other has different with dst?
+    bool is_nhwc = dst.get_desc().is_nhwc();
+    bool is_ndhwc = dst.get_desc().is_ndhwc();
+    auto format_tag = is_nhwc ? tag::nhwc : (is_ndhwc ? tag::ndhwc : tag::any);
+    auto other_desc = other.get_desc().to_format(format_tag);
     auto& expected_other = reorder_src ?
-        other.reorder_if_differ_in(pd.dst_desc()) :
+        other.reorder_if_differ_in(other_desc) :
```

Comment:
we sure there is no perf impact from this extra reorder?

Reply:
In PyTorch mkldnn_convolution_pointwise_binary kernel, we have converted both input and other to ChannelsLast format. Other's format will not be different from dst's here, and no reorder will be done. But I'm not sure whether other will be different from dst in other usage scenarios, so I still keep this line of code to make sure other has same format with dst.